### PR TITLE
Refactor CommentCount

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR lifts up the call to get the count of comments for an article up out of `Counts` and into `App`

## Why?
Because we also want to use this count elsewhere on the page. The `SignedInAs` component needs it and we will also want to pass it to Comments later when we load them. 

## Link to supporting Trello card
https://trello.com/c/QmSt9xB6/1264-refactor-counts

### CODE REVIEW

File: src/web/browser/App.tsx [hunk 1]
```diff
@@ -58,6 +60,17 @@ const App = ({ CAPI, NAV }: Props) => {
...
+    useEffect(() => {
+        const callFetch = async () =>
+            setCommentCount(
+                await getCommentCount(
+                    CAPI.config.ajaxUrl,
+                    CAPI.config.shortUrlId,
+                ),
+            );
+        callFetch();
+    }, [CAPI.config.ajaxUrl, CAPI.config.shortUrlId]);
```
Comment (liywjl) [inline src/web/browser/App.tsx]: we want to move away from `CAPI.config` I understand it should be refactored in another PR, so I guess its ok

File: src/web/lib/getCommentCount.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return fetch(url)
+        .then(response => {
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response;
+        })
+        .then(response => response.json())
+        .then(
+            json =>
```
Comment (liywjl) [inline src/web/lib/getCommentCount.ts]: can we not use typescript `json?.counts?[0] || 0` ?

File: src/web/browser/App.tsx (hunk 1 above)
Comment (oliverlloyd) [reply src/web/browser/App.tsx]: Yeah, I thought that when using it but backed off changing it in favour of doing the work later when relevant

File: src/web/lib/getCommentCount.ts (hunk 2 above)
Comment (oliverlloyd) [reply src/web/lib/getCommentCount.ts]: I wish but we still get linting snags around this
